scripts/obtain/dotnet-install.sh

```diff
@@ -765,7 +766,7 @@ do
             override_non_versioned_files=false
             ;;
         -?|--?|-h|--help|-[Hh]elp)
-            script_name="$(basename $0)"
+            script_name="$(basename "$0")"
```

Comment:
Is this necessary: "$0"? What is the breaking scenario?

Reply:
Hypothetically:
```
$ basename music/imagine dragons
imagine
$ basename "music/imagine dragons"
imagine dragons
$ basename a dir/file
a
$ basename "a dir/file"
file
```

If the script name **never** contains spaces, then there wouldn't be any issues. If the full path to the script **never** contains spaces, then there wouldn't be any issues. The goal was to fix potential space-containing paths breaking scripts.

Reply:
So it's the path that could potentially contains spaces; okay - good.